autogluon/utils/tabular/ml/models/abstract/model_trial.py

```diff
@@ -8,6 +8,7 @@
 
 logger = logging.getLogger(__name__)  # TODO: Currently unused
 
+
 @args()
 def model_trial(args, reporter):
```

Comment:
LightGBM and NN have their own custom trial functions for HPO that this would also need to be applied to:

utils/tabular/ml/models/lgb/hyperparameters/lgb_trial.py
utils/tabular/ml/models/tabular_nn/tabular_nn_trial.py

Looking at these very old functions, we will probably want to refactor them a bit beyond this HPO fix (though not in this PR) to be less disjointed from the usual .fit function.

Reply:
I added for GBM and NN.

Yep, you are right, they are old and a bit ugly :)

Reply:
Meanwhile I would like to ask, why there is no HPO support for RF and KNN and XT ? Do we plan to have them in the feature ?

Reply:
It is quite easy to add, but at the time HPO was unstable enough such that I didn't want to encourage such use. If it appears to be stable after this fix, I can quickly add the functionality in.

A more complex point is that some of these models might not benefit from HPO as much as the others, and would be taking valuable training time away from the other models if they were tuned.

In terms of disk space, RF, XT, and KNN take up much more space than the other models, so I didn't want AutoGluon to be taking 100+ GB of space while hyperparameter tuning, since we save every model (KNN can easily take up 10MB+ of space/second during HPO on big datasets) This is probably the biggest concern for me, I don't want people to run out of disk space using AutoGluon. We may have to check remaining disk space and ensure we don't go below some threshold.